semantic-core/semantic-core.cabal

```diff
@@ -53,7 +54,6 @@ library
     Core.Core.Parser
     Core.Core.Pretty
     Core.Eval
-    Core.File
```

Comment:
This PR is super straightforward, I just renamed the one file because it relates to the analysis of projects more so than to `Core` itself.